pipeline_dp/dataframes.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2023 OpenMined.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Computing DP aggregations on (Pandas, Spark, Beam) Dataframes."""
+
+from dataclasses import dataclass
+from typing import Any, Dict, Iterable, List, Optional
+
+import pipeline_dp
+
+
+@dataclass
+class Columns:
+    privacy_key: str
+    partition_key: str
+    value: Optional[str]
+
+
+@dataclass
+class ContributionBounds:
+    max_partitions_contributed: Optional[int] = None
+    max_contributions_per_partition: Optional[int] = None
+    min_value: Optional[float] = None
+    max_value: Optional[float] = None
+
+
+@dataclass
+class Budget:
+    epsilon: float
+    delta: float = 0
+
+    def __post_init__(self):
+        if self.epsilon <= 0:
+            raise ValueError(f"epsilon must be positive, not {self.epsilon}.")
+        if self.delta < 0:
+            raise ValueError(f"delta must be non-negative, not {self.delta}.")
+
+
+class Query:
+    """DP Query on DataFrames.
+
+    Warning: Don't create it directly, use QueryBuilder().
+    """
+
+    def __init__(self, df, columns: Columns, metrics: Dict[pipeline_dp.Metric,
```

Comment:
`self._metrics = {}  # map from pipeline_dp.Metric -> output column name` from below.
Then metrics here should be of type `Dict[Metric, str]`.

Reply:
Yes, thanks